src/configuration.rs

```diff
@@ -0,0 +1,376 @@
+         * 
+        // Always encrypt the seed w/ the derived "blinding" key; the AEAD "tag" (Nonce, MAC and
+        // size) is prefixed to the resultant encrypted seed.
+        let seed_nonce: [u8; 8] = rand::random();
+        let seed_aad = [0u8; 0]; // additional authenticated data
+        let seed_key = blind_keypair.secret.to_bytes();
+        let mut seed_encrypted = [0u8; HOLO_ENTROPY_SIZE];
+        let mut seed_tag = [0u8; AEAD_TAGBYTES];
+        ChaCha20Poly1305::new(&seed_key, &seed_nonce, &seed_aad)
+            .encrypt(&seed_entropy, &mut seed_encrypted, &mut seed_tag);
+        let mut seed_cipher = [0u8; AEAD_TAGBYTES + HOLO_ENTROPY_SIZE];
+        *array_mut_ref![seed_cipher, 0, AEAD_TAGBYTES] = *array_ref![seed_tag, 0, AEAD_TAGBYTES];
+        *array_mut_ref![seed_cipher, AEAD_TAGBYTES, HOLO_ENTROPY_SIZE] = *array_ref![seed_encrypted, 0, HOLO_ENTROPY_SIZE];
+         * 
+         */
+        // To use AES CBC, we could use the admin public key as the IV.  The admin keypair, the
+        // derived blinding key, and the seed entropy are all created simultaneously, and the
+        // blinding key is used exactly once, so the requirement that the AES CBC IV be
+        // "unpredictable prior to use" is satisfied.  We don't have to come up with subsequent IVs,
+        // because we are using the key only one time.  We can also use AES ECB w/ no IV, because we
+        // are producing exactly one cipher-block of output.
+        let seed_key = blind_keypair.secret.to_bytes();
+        let seed_cipher = encrypt_seed_with(&seed_entropy, &seed_key)?;
+
+        let config = HoloPortConfiguration {
+            name:		name_maybe.to_owned(),
+            email:		email.to_string(),
+            admin_pubkey:	hcid::HcidEncoding::with_kind("hca0")?.encode(&admin_keypair.public.to_bytes())?,
+            seed:		hcid::HcidEncoding::with_kind("hcc0")?.encode(&seed_cipher)?,
+            seed_sig:		base64::encode(&seed_sig.to_bytes().into_iter()),
+            seed_key:		Some(hcid::HcidEncoding::with_kind("hcb0")?.encode(&seed_key)?),
+        };
+
+        // Final test to guarantee round-trip correctness
+        if decrypt_seed_with(
+            &config.admin_pubkey,		// HcA...  Holo Admin key
+            &config.seed_key.clone().unwrap(),	// HcB...  Holo Blinding key
+            &config.seed,			// HcC...  Holo Config seed (Encrypted w/ Blinding key)
+            &config.seed_sig,			// Signature of unencrypted seed entropy
+        )? != seed_entropy {
+            return Err(ConfigurationError::Generic(format!(
+                "Unable to recover seed entropy of config: {:?}", config
+            )))
+        };
+        Ok(config)
+    }
+}
+
+/// encrypt_seed_with -- sign with admin_and encrypt the seed with the given seed "blinding" key
+pub fn encrypt_seed_with(
+    seed: &[u8],
+    seed_key: &[u8]
+) -> Result<Vec<u8>, ConfigurationError> {
+    let mut seed_enc = aes::ecb_encryptor(
```

Comment:
https://blog.filippo.io/the-ecb-penguin/

ECB implementation that you're using here seems to use 16 byte block size, which means this gives away distance between first and second blocks of seed (which is 32 bytes long).

Reply:
Nope, I'm using 256-bit AES: exactly one cipher block will be used.  I realize that multiple blocks leak information, but by using one block, we don't have to have an IV or chaining, to avoid leaking any information.  If multiple blocks were used, we'd have that problem.

Reply:
Right. But at this point, you could use a one-time pad instead.